### DISCUSSION TITLE
Meant for scaffolding once and never again or incrementally?

### DISCUSSION THREAD

**Discussion by NoTuxNoBux** [pdevito3/craftsman#110] [Q&A] [answered]
I'm completely new to Wrapt, and a bit confused about whether or not it is intended to be used pretty much once and then never again after customizing, or can actually incrementally add new properties and generate migrations for you.

[The customizing Wrapt projects documentation page says the following:](https://wrapt.dev/docs/customizing-wrapt-projects)

> For example, if you go into an Entity file, you'll notice a comment at the bottom like this // add-on property marker - Do Not Delete This Comment. This comment is what Craftsman will look for when adding new properties to your entity, so if it were removed, that operation would no longer be able to run properly.

But there doesn't appear to be any `craftsman update entity foo.yaml`. If you run `add` a second time, it just errors out that the domain entities already exist. I can of course add new entities without issue, but I can't add new properties to existing ones and have Wrapt update them without dropping all the existing code (including customization) first.

So the documentation implies Wrapt can be incremental and work with existing generated code (hence the markers), but is it not actually implemented or am I just missing something?

**Comment by pdevito3** [comment, marked as answer]
Scaffolding files are meant to be one time use and then thrown away, not act as a source of truth. Once you've scaffolded an initial project, you can do something like 'add:entity' with an appropriate file to continue to add and extend the project but the scaffolding file is just meant to tell craftsman what boilerplate to add for you and then be thrown away. 

The section you mentioned about the property comment is out of date. That option was removed since adding properties wasn't a big value add for this and could be done fine by hand. I might add a flavor of it back eventually to better support FKs but for now it's not in

Does that make sense?

**Reply by NoTuxNoBux** [reply]
Yes, that makes sense, and was what I initially thought Wrapt was for :+1: - I just got confused after seeing the note in the documentation. 'Scaffolding' is sometimes more broadly used, as well - e.g. Keystone (JavaScript) 'scaffolds' dynamically and can work incrementally as well.

Thanks for the quick response!

**Reply by pdevito3** [reply]
Yeah that's fair, no worries. Welcome to the community! Holler if you have any issues or questions.
